service/matching/handler/engine.go

```diff
@@ -872,22 +872,30 @@ func (e *matchingEngineImpl) QueryWorkflow(
 
 	taskID := uuid.New()
 	resp, err := tlMgr.DispatchQueryTask(hCtx.Context, taskID, queryRequest)
```

Comment:
Should the TaskListManager return a response that already has the TaskListPartitionConfig instead? It seems conceptually a little cleaner, since the TaskListManager manages both.

Reply:
The partition config in the response is from a forwarded response which is not from this partition. I want the response to include the partition config from this partition.

Reply:
Nit: I still think that's something the TaskListManager should be responsible for updating rather than the matching engine.